pkg/controller/windowsmachineconfig/nodeconfig/network.go

```diff
@@ -0,0 +1,137 @@
+package nodeconfig
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+)
+
+// cniConf contains the structure of the CNI template
+type cniConf struct {
+	CNIVersion   string       `json:"cniVersion"`
+	Name         string       `json:"name"`
+	Type         string       `json:"type"`
+	Capabilities capabilities `json:"capabilities"`
+	IPAM         ipam         `json:"ipam"`
+	Policies     policies     `json:"policies"`
+}
+
+// nested structs required for creating
+// cniConf struct
+type capabilities struct {
+	DNS bool `json:"dns"`
+}
+
+type ipam struct {
+	Type   string `json:"type"`
+	Subnet string `json:"subnet"`
+}
+
+type policies []struct {
+	Name  string `json:"name"`
+	Value value  `json:"value"`
+}
+
+type value struct {
+	Type              string   `json:"Type"`
+	ExceptionList     []string `json:"ExceptionList,omitempty"`
+	DestinationPrefix string   `json:"DestinationPrefix,omitempty"`
+	NeedEncap         bool     `json:"NeedEncap"`
+}
+
+// network struct contains the node network information
+type network struct {
+	// hostSubnet holds the node host subnet value
+	hostSubnet string
+}
+
+// newNetwork returns a pointer to the network struct
+func newNetwork() *network {
+	return &network{}
+}
+
+// setHostSubnet sets the value for hostSubnet field in the
+// network struct
+func (nw *network) setHostSubnet(hostSubnet string) error {
+	if hostSubnet == "" || validateCIDR(hostSubnet) != nil {
+		return errors.Errorf("error receiving valid value for node hostSubnet")
+	}
+	nw.hostSubnet = hostSubnet
+	return nil
+}
+
+// validateCIDR uses the parseCIDR from network package
+// to validate the format of the CIDR received
+func validateCIDR(serviceCIDR string) error {
+	_, _, err := net.ParseCIDR(serviceCIDR)
+	if err != nil || serviceCIDR == "" {
+		return errors.Wrapf(err, "error receiving valid CIDR value")
+	}
+	return nil
+}
+
+// populateCniConfig populates the CNI config template with necessary information and
+// creates a new file in temp directory to store the modified template
+func (nw *network) populateCniConfig(serviceCIDR string, templatePath string) (string, error) {
+	if nw.hostSubnet == "" {
+		return "", errors.New("received empty value for node hostSubnet, can't " +
+			"populate CNI config template")
+	}
+
+	if validateCIDR(serviceCIDR) != nil || serviceCIDR == "" {
```

Comment:
Please swap the checks. Moreover this is what I meant by guaranteeing service CIDR values. `network` is local to `nodeConfig` and not exported. So as long as you guarantee that the service CIDR passed to `nodeConfig` is valid, you don't need to check here. This needs to be added here only if `network` is exported for use outside of `nodeConfig`.

Reply:
got it will remove the validate check from here since we are validating it while creating the new node config.